src/components/FormElements/ResourceFeePickerWithQuery.tsx

```diff
@@ -0,0 +1,185 @@
+// the minimum resource fee
+export const ResourceFeePickerWithQuery = ({
+  id,
+  labelSuffix,
+  label,
+  value,
+  error,
+  onChange,
+  placeholder,
+  note,
+  infoLink,
+  disabled,
+  ...props
+}: ResourceFeePickerWithQueryProps) => {
+  const { network, transaction } = useStore();
+  const { params: txnParams, soroban, isValid } = transaction.build;
+  const { operation } = soroban;
+  const {
+    mutateAsync: simulateTx,
+    data: simulateTxData,
+    isPending: isSimulateTxPending,
+  } = useSimulateTx();
+
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(
+    undefined,
+  );
+
+  // operation.params.resource_fee is required for submitting; however,
+  // we don't check for it here in case user doesn't have one and we still want to simulate
+  // @TODO should update this from isValid.operations point of view
+  // It doesn't validate the input at the moment
+  const isOperationValidForSimulateTx = isAllParamsExceptResourceFeeValid(
+    operation.params,
+  );
+
+  useEffect(() => {
+    // If the user has fetched the min resource fee, update the input's value
+    if (simulateTxData?.result?.minResourceFee) {
+      const syntheticEvent = {
+        target: {
+          value: simulateTxData.result.minResourceFee.toString(),
+        },
+      } as React.ChangeEvent<HTMLInputElement>;
+
+      onChange(syntheticEvent);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [simulateTxData?.result?.minResourceFee]);
+
+  // Create a sample transaction to simulate to get the min resource fee
+  const buildTxToSimulate = () => {
+    try {
+      // We add a bogus fee to simualte to fetch the min resource fee from the RPC
+      const BOGUS_RESOURCE_FEE = "100";
+
+      let builtXdr, contractDataXDR;
+
+      try {
+        contractDataXDR = getContractDataXDR({
+          contractAddress: operation.params.contract,
+          dataKey: operation.params.key_xdr,
+          durability: operation.params.durability,
+        });
+      } catch (e) {
+        throw new Error(`Failed to generate contract data XDR: ${e}`);
+      }
+
+      if (!contractDataXDR) {
+        throw new Error("Failed to fetch contract data XDR");
+      }
+
+      const sorobanData = getSorobanTxData({
+        contractDataXDR,
+        operationType: operation.operation_type as SorobanOpType,
+        fee: BOGUS_RESOURCE_FEE, // simulate purpose only
+      });
+
+      if (sorobanData) {
+        builtXdr = buildSorobanTx({
+          sorobanData,
+          params: txnParams,
+          sorobanOp: {
+            ...operation,
+            params: {
+              ...operation.params,
+              resource_fee: BOGUS_RESOURCE_FEE,
+            },
+          },
+          networkPassphrase: network.passphrase,
+        }).toXDR();
+      } else {
+        throw new Error("Failed to build Soroban transaction data");
+      }
+      return builtXdr;
+    } catch (e) {
+      setErrorMessage(`${e}`);
```

Comment:
Should the final message note that this is for the fee? I'm not sure if it would be clear to the user that these error messages relate to that.